src/app/onboarding/availability/page.tsx

```diff
@@ -1,24 +1,106 @@
 'use client';
 
-import Button from '@/components/Button';
-import COLORS from '@/styles/colors';
+import { useContext, useMemo, useState, useEffect } from 'react';
 import { H1 } from '@/styles/text';
 import { OnboardingContext } from '@/utils/OnboardingProvider';
-import { useContext } from 'react';
+import TextInput from '@/components/TextInput';
+import { isValidDate } from '@/utils/helpers';
+import DateInput from '@/components/DateInput';
+import TextAreaInput from '@/components/TextAreaInput';
 
 export default function Page() {
   const onboarding = useContext(OnboardingContext);
+  const [hours, setHours] = useState('');
+  const [startDate, setStartDate] = useState('');
+  const [periods, setPeriods] = useState('');
+
+  const getHoursErrorText = useMemo(
+    () =>
+      hours && Number.isNaN(parseInt(hours, 10)) ? 'Must be a number' : '',
+    [hours],
+  );
+
+  const getStartDateErrorText = useMemo(
+    () =>
+      startDate && !isValidDate(startDate)
+        ? 'Must select a current or future date'
+        : '',
+    [hours],
+  );
+
+  useEffect(() => {
+    setHours(
+      onboarding?.profile.hours_per_month
+        ? onboarding?.profile.hours_per_month.toString()
+        : '',
+    );
+    setStartDate(onboarding?.profile.start_date || '');
+    setPeriods(onboarding?.profile.availability_description || '');
+
+    if (
+      onboarding?.profile.hours_per_month &&
+      onboarding?.profile.hours_per_month > 0 &&
+      onboarding?.profile.start_date &&
+      isValidDate(onboarding?.profile.start_date)
+    ) {
+      onboarding?.setCanContinue(true);
+    } else {
+      onboarding?.setProgress(2);
+      onboarding?.setCanContinue(false);
+    }
+  }, [onboarding]);
+
+  const handleHours = (v: string) => {
+    if (!Number.isNaN(parseInt(v, 10))) {
+      onboarding?.updateProfile({
+        hours_per_month: +v,
+      });
+    }
+  };
+
+  const handleStartDate = (v: string) => {
+    if (v === '' || isValidDate(v)) {
+      onboarding?.updateProfile({
+        start_date: v,
+      });
+    }
+  };
```

Comment:
we shouldn't add checks to the update. if you delete a field, you can move forward, and when you navigate back, it will not have changed. It should update regardless of value, and we'll know it's correct based on whether or not they can continue.

Reply:
in that case, we may want to add a `useState`